src/components/nav/nav-item-dropdown.js

```diff
@@ -41,57 +47,62 @@ export const BNavItemDropdown = /*#__PURE__*/ Vue.extend({
     }
   },
   render(h) {
-    const button = h(
+    const { toggleId, visible } = this
+
+    const $toggle = h(
       BLink,
       {
-        ref: 'toggle',
         staticClass: 'nav-link dropdown-toggle',
         class: this.toggleClasses,
         props: {
-          href: '#',
+          href: this.href || `#${toggleId || ''}`,
```

Comment:
use: 
```js
href: `#${this.id || ''}`,
```

or 

```js
href: `#${toString(this.id)}`,
```

Reply:
This way the `href` is only set to something other than `#` when the user supplied an ID.  The outer `<li>` will have this ID if a user provide one.

Reply:
And then we can get rid of the `href` prop altogether.